### PR TITLE
In 151774703 standardize redux actions

### PR DESCRIPTION
**Pull request by NateOSI**
### Pivotal Story

- [**TECHNICAL STORY** Standardize redux actions 151774703](https://www.pivotaltracker.com/story/show/151774703)

### Purpose
As a **Developer** 
**I need** to be able to make certain assumptions about how
data is messaged throughout my redux application.
**So that** I can design consistent code that is readable

### CODE REVIEW

**Comment by TylerOSI** [comment]
are there not supposed to be failure actions with FSA compliant actions? I thought the point was to reduce the number of actions by including a payload and a status with the actions?

File: app/javascript/actions/contactActions.js [hunk 1]
```diff
@@ -5,20 +5,59 @@ export const FETCH_CONTACT = 'FETCH_CONTACT'
...
+      note,
+      purpose,
+      communication_method,
+      location,
+    },
+  }
 }
 export function saveFailure(error) {
-  return {type: SAVE_CONTACT_FAILURE, error}
+  return {type: SAVE_CONTACT_FAILURE, payload: {error}}
```
Comment (jgedeon-osi) [inline app/javascript/actions/contactActions.js]: Am I correct in assuming that you are tackling adding payload to all the actions first before joining the FAILURE and SUCCESS actions into one?

File: app/javascript/actions/httpActions.js [hunk 2]
```diff
@@ -1,7 +1,5 @@
-import {
-  HTTP_FAILURE,
-} from 'actions/actionTypes'
+import {HTTP_FAILURE} from 'actions/actionTypes'
 
 export function httpError(responseJSON) {
-  return {type: HTTP_FAILURE, responseJSON}
+  return {type: HTTP_FAILURE, payload: responseJSON}
```
Comment (jgedeon-osi) [inline app/javascript/actions/httpActions.js]: FYI the error banner story will affect these. Who ever is first will have a conflict.

File: app/javascript/reducers/crossReportFormReducer.js [hunk 3]
```diff
@@ -107,16 +107,16 @@ export default createReducer(Map(), {
...
   },
-  [TOUCH_CROSS_REPORT_FIELD](state, {field}) {
+  [TOUCH_CROSS_REPORT_FIELD](state, {payload: {field}}) {
     return state.setIn([field, 'touched'], true)
   },
-  [TOUCH_CROSS_REPORT_AGENCY_FIELD](state, {agencyType}) {
+  [TOUCH_CROSS_REPORT_AGENCY_FIELD](state, {payload: {agencyType}}) {
     return state.setIn([agencyType, 'agency', 'touched'], true)
   },
   [FETCH_SCREENING_SUCCESS](_state, {screening}) {
```
Comment (jgedeon-osi) [inline app/javascript/reducers/crossReportFormReducer.js]: Did you mean to skip FETCH_SCREENING_SUCCESS?

Comment (NateOSI) [reply app/javascript/reducers/crossReportFormReducer.js]: yeah still need to do this

File: app/javascript/home/HomePage.jsx [hunk 4]
```diff
@@ -41,20 +35,19 @@ export class HomePage extends React.Component {
...
-  screening: PropTypes.object.isRequired,
+  screenings: PropTypes.array,
 }
 
-function mapStateToProps(state, ownProps) {
+function mapStateToProps(state, _ownProps) {
   return {
-    screening: state.get('screening'),
-    router: ownProps.router,
+    screenings: state.get('screenings').toJS(),
```
Comment (saraOSI) [inline app/javascript/home/HomePage.jsx]: Woohoo!

**Pull request merged**